_internal/utils/sourceme.R

```diff
@@ -206,9 +61,19 @@ restore_and_run <- function(app_name, username = "insightsengineering", repo = "
   #'
   #' @param app_directory The directory path of the Shiny app.
   #' @param ... Additional arguments passed to `shiny::runApp()`.
-  load_and_run_app <- function(app_directory, ...) {
-    renv::load(app_directory)
-    renv::restore(clean = TRUE, prompt = FALSE)
+  load_and_run_app <- function(app_directory, package_repo, ...) {
+    renv::dependencies()$Package |>
+      tools::package_dependencies() |>
+      unlist() |>
+      unname() |>
+      unique() |>
+      install.packages(
+        repos = c(
+          Pharmaverse = package_repo,
+          CRAN = "https://cloud.r-project.org",
+          BioC = BiocManager::repositories()
+        )
+      )
```

Comment:
This step has a significant side effect meaning that after running `load_and_run_app` my environment is not the same as before run - I have different set of libraries installed. It's a significant limitation over renv-based approach which leaves the outside world untouched.

Reply:
Yeah, it was a hard choice. While I like the fact that this makes sure all the teal packages are updated I dislike that this will also update all the cran packages like ggplot2. One small advantage I see with this implementation is that it does not that that long to install the dependencies compared to renv.

I was biased by the _"CI should always deploy based on the latest packages"_. But perhaps the CI and the user run should be different and we should still have `renv::restore()`. We could also have both of these ways of restoring packages and mention this to the user. WDYT? How shall we proceed?

Reply:
I feel that what you wanted to do is to add new repos from where we can install our packages. What you actually did is to change `load_and_run` (that installs deps into app' envir) into `install_and_run` (that installs deps into user envir). This is a significant change in the logic and I would say it's a no-go. We don't want to have such a side effect. Such requirement can be also handled inside renv.

My recommendation is to still use renv because it will keep app' packages separate from user packages. It's safer. Of course you can continue current way and install to temp libpath but I feel that renv will be safer. It has repos field embedded in the lock file structure and it will probably solve many other issues that we are not anticipating now.
Even though you can solve this multiple ways, for the maintenance reasons I would like to have only one way, i.e. user and CI run as similar as possible and any differences should be handled inside workflow definition.

When it comes to the CI, we are stepping a little bit into https://github.com/insightsengineering/idr-tasks/issues/580 and I have following idea:
- stable deploy:
  - `renv::update()` using stable repo
  - autocommit diff to lock file
  - test
  - deploy
- devel deploy
  - (same as above but using devel repo and without autocommit)